org.eclipse.draw2d/src/org/eclipse/draw2d/XYLayout.java

```diff
@@ -136,7 +136,11 @@ public void remove(IFigure figure) {
 	@Override
 	public void setConstraint(IFigure figure, Object newConstraint) {
 		super.setConstraint(figure, newConstraint);
-		if (newConstraint instanceof Rectangle rect) {
+		if (newConstraint != null) {
+			if (!(newConstraint instanceof Rectangle rect)) {
```

Comment:
That nested if looks nasty. And wouldn't it be better to throw the exception before calling super.setConstraint()?

I.e.
```
if (newConstraint instanceof Rectangle) {
  ....
} else {
  throw new IllegalArgumentException()
}
```

Reply:
` newConstraint` can be null and should then be ignored. At least this was how the code worked before.

Reply:
Good point...